File: src/com/eagle/programmar/Javascript/Functions/Javascript_LambdaFunction.java
// Copyright Eagle Legacy Modernization, 2010-date
// Original author: Steven A. O'Hara, Apr 1, 2024

package com.eagle.programmar.Javascript.Functions;

import com.eagle.programmar.Javascript.Javascript_Element;
import com.eagle.programmar.Javascript.Javascript_FunctionBody;
import com.eagle.programmar.Javascript.Symbols.Javascript_Variable_Definition;
import com.eagle.programmar.Javascript.Terminals.Javascript_Keyword;
import com.eagle.programmar.Javascript.Terminals.Javascript_Punctuation;
import com.eagle.tokens.PrimaryOperator;
import com.eagle.tokens.SeparatedList;
import com.eagle.tokens.TokenChooser;
import com.eagle.tokens.TokenSequence;
import com.eagle.tokens.punctuation.PunctuationComma;
import com.eagle.tokens.punctuation.PunctuationLeftParen;
import com.eagle.tokens.punctuation.PunctuationRightParen;

public class Javascript_LambdaFunction extends PrimaryOperator
{
	public @S(10) @OPT Javascript_Keyword ASYNC = new Javascript_Keyword("async");
	public @S(20) Javascript_LambdaParams params;
	public @S(30) Javascript_Punctuation arrow = new Javascript_Punctuation("=>");
	public @S(40) Javascript_LambdaBody body;

	public static class Javascript_LambdaParams extends TokenChooser
	{
		public @CHOICE Javascript_Variable_Definition XXparam;

		public @CHOICE static class Javascript_LambdaManyParams extends TokenSequence
		{
			public @S(10) PunctuationLeftParen leftParen;
			public @S(20) @OPT SeparatedList<Javascript_Variable_Definition, PunctuationComma> params;
			public @S(30) PunctuationRightParen rightParen;
		}
	}

	public static class Javascript_LambdaBody extends TokenChooser
	{
		public @CHOICE Javascript_FunctionBody XXblock;
		public @CHOICE Javascript_Element XXstmt;
	}
}

File: csrc/com/eagle/programmar/Javascript/Functions/Javascript_LambdaFunction.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization, 2010-date
// Original author: Steven A. O'Hara, Apr 1, 2024

namespace com.eagle.programmar.Javascript.Functions
{
	using Javascript_Element = com.eagle.programmar.Javascript.Javascript_Element;
	using Javascript_FunctionBody = com.eagle.programmar.Javascript.Javascript_FunctionBody;
	using Javascript_Variable_Definition = com.eagle.programmar.Javascript.Symbols.Javascript_Variable_Definition;
	using Javascript_Keyword = com.eagle.programmar.Javascript.Terminals.Javascript_Keyword;
	using Javascript_Punctuation = com.eagle.programmar.Javascript.Terminals.Javascript_Punctuation;
	using PrimaryOperator = com.eagle.tokens.PrimaryOperator;
	using SeparatedList = com.eagle.tokens.SeparatedList;
	using TokenChooser = com.eagle.tokens.TokenChooser;
	using TokenSequence = com.eagle.tokens.TokenSequence;
	using PunctuationComma = com.eagle.tokens.punctuation.PunctuationComma;
	using PunctuationLeftParen = com.eagle.tokens.punctuation.PunctuationLeftParen;
	using PunctuationRightParen = com.eagle.tokens.punctuation.PunctuationRightParen;

	public class Javascript_LambdaFunction : PrimaryOperator
	{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) @OPT Javascript_Keyword ASYNC = new com.eagle.programmar.Javascript.Terminals.Javascript_Keyword("async");
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) Javascript_LambdaParams params;
		public Javascript_LambdaParams @params;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.programmar.Javascript.Terminals.Javascript_Punctuation arrow = new com.eagle.programmar.Javascript.Terminals.Javascript_Punctuation("=>");
		public Javascript_Punctuation arrow = new Javascript_Punctuation("=>");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) Javascript_LambdaBody body;
		public Javascript_LambdaBody body;

		public class Javascript_LambdaParams : TokenChooser
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE Javascript_Variable_Definition XXparam;
			public Javascript_Variable_Definition XXparam;

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE static class Javascript_LambdaManyParams extends com.eagle.tokens.TokenSequence
			public class Javascript_LambdaManyParams : TokenSequence
			{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.tokens.punctuation.PunctuationLeftParen leftParen;
				public PunctuationLeftParen leftParen;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) @OPT SeparatedList<com.eagle.programmar.Javascript.Symbols.Javascript_Variable_Definition, com.eagle.tokens.punctuation.PunctuationComma> params;
				public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.tokens.punctuation.PunctuationRightParen rightParen;
				public PunctuationRightParen rightParen;
			}
		}

		public class Javascript_LambdaBody : TokenChooser
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE Javascript_FunctionBody XXblock;
			public Javascript_FunctionBody XXblock;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE Javascript_Element XXstmt;
			public Javascript_Element XXstmt;
		}
	}

}
